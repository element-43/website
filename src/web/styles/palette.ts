export default {
    brand: {
        // Primary.
        purple300: '#9a6dea',
        purple400: '#8154d1',
        purple500: '#673ab7',
        purple600: '#4e219e',
        purple700: '#340784',
        purple800: '#1b006b',

        // Accent.
        green300: '#61ffa4',
        green400: '#48e68b',
        green500: '#2ecc71',
        green600: '#15b358',
        green700: '#00993e',
        green800: '#008025',
    },

    greyScale: {
        white: '#fff',
        lightGrey: '#f2f2f2',
        grey: '#ccc',
        darkGrey: '#8c8c8c',
        black: '#000',
    },

    ui: {
        darkRed: '#8b0000',
        orange: '#ffa500',
        orangeRed: '#ff4500',
        pink: '##ff69b4',
        red: '#cc0000',
        salmon: '#fa8072',
        slateGrey: '#708090',
        yellow: '#ffff00',
    },
};
