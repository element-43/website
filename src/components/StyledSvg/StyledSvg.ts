import styled from 'styled-components';

// Theme.
import palette from '../../theme/palette';

export interface IProps {
  color?: string;
  hoverColor?: string;
  size?: string;
  viewBox: string;
}

export const StyledSvg = styled.svg<IProps>`
  height: ${(props) => props.size || '1rem'};

  ${(props) =>
    props.hoverColor &&
    `
    &:hover {
      path {
        fill: ${props.hoverColor};
      }
    }
  `}

  path {
    fill: ${(props: IProps) => props.color || palette.greyScale.black};
    ${(props: IProps) =>
      props.hoverColor &&
      `
        transition: fill 0.3s ease-in-out;
    `}
  }
`;

export default StyledSvg;
