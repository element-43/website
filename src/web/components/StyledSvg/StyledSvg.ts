import styled from 'styled-components';

// Styles.
import palette from '../../styles/palette';

export interface Props {
  color?: string;
  hoverColor?: string;
  size?: string;
  viewBox: string;
}

export const StyledSvg = styled.svg<Props>`
  height: ${(props) => props.size || '1rem'};

  ${(props) =>
    props.hoverColor &&
    `
    &:hover {
      path {
        fill: ${props.hoverColor};
      }
    }
  `}

  path {
    fill: ${(props: Props) => props.color || palette.greyScale.black};
    ${(props: Props) =>
      props.hoverColor &&
      `
        transition: fill 0.3s ease-in-out;
    `}
  }
`;

export default StyledSvg;
