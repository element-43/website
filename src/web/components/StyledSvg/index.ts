import styled from 'styled-components';

// Styles.
import palette from '../../styles/palette';

interface Props {
    color?: string;
    hoverColor?: string;
    size?: string;
    viewBox: string;
}

const StyledSvg = styled<Props, 'svg'>('svg')`
  height: ${props => props.size || '1rem'};
  
  ${props => props.hoverColor && `
    &:hover {
      path {
        fill: ${props.hoverColor};
      }
    }
  `}
  
  path {
    fill: ${props => props.color || palette.greyScale.black};
    ${props => props.hoverColor && `
        transition: fill 0.3s ease-in-out;
    `}
  }
`;

export {
    StyledSvg,
    Props
};
