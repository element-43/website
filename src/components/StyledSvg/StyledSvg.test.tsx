import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

// Config.
import palette from '../../theme/palette';

// Components.
import { Props, StyledSvg } from './StyledSvg';

interface IScope {
  props: Props;
  wrapper: ShallowWrapper;
}

describe('<StyledSvg />>', () => {
  let scope: IScope;

  beforeEach(() => {
    const props: Props = {
      viewBox: '0 0 0 0',
    };

    scope = {
      props,
      wrapper: shallow(<StyledSvg {...props} />),
    };
  });

  describe('<StyledSvg /> snapshots', () => {
    it('should match the snapshot with the default props', () => {
      expect(scope.wrapper).toMatchSnapshot();
    });

    it('should match the snapshot with the optional props', () => {
      scope.props.color = palette.ui.red;
      scope.props.hoverColor = palette.greyScale.white;
      scope.props.size = '100px';

      expect(shallow(<StyledSvg {...scope.props} />)).toMatchSnapshot();
    });
  });
});
