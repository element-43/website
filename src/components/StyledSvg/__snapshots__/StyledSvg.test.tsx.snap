// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledSvg />> <StyledSvg /> snapshots should match the snapshot with the default props 1`] = `ShallowWrapper {}`;

exports[`<StyledSvg />> <StyledSvg /> snapshots should match the snapshot with the optional props 1`] = `ShallowWrapper {}`;
